### PR TITLE
fix TODO for setting hostport in the kubeconfig file for the host

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR fixes a TODO related to HA (sorry I missed that in the past)

More specifically, when copying the kubeconfig file to the host, now kind uses the load balancer `haproxy.ControlPlanePort` in case a load balancer exists, otherwise kind uses the control-plane `kubeadm.APIServerPort` (the latter is the only case managed ATM 😞)

/assign @neolit123  @BenTheElder
/priority important-soon
/kind cleanup

### CODE REVIEW

File: pkg/cluster/internal/create/kubeadm-init.go [hunk 1]
```diff
@@ -155,6 +154,29 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
...
+func getHostPort(ec *execContext, controlPlane *nodes.Node) (int, error) {
+	// in single control-plane cluster, the node exposing the API server to the outside world is the control-plane itself
+	node := controlPlane
+	port := kubeadm.APIServerPort
+
+	// in case an lb is defined, use the lb node instead
+	if ec.ExternalLoadBalancer() != nil {
+		loadBalancerHandle, ok := ec.NodeFor(ec.ExternalLoadBalancer())
+		if !ok {
+			return 0, errors.Errorf("unable to get the handle for operating on node: %s", ec.ExternalLoadBalancer().Name)
```
Comment (neolit123) [inline pkg/cluster/internal/create/kubeadm-init.go]: optional nit: `unable to get the handle of load balancer node: %s` ?

File: pkg/cluster/internal/create/kubeadm-init.go [hunk 2]
```diff
@@ -155,6 +154,29 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
...
+		}
+
+		node = loadBalancerHandle
+		port = haproxy.ControlPlanePort
```
Comment (neolit123) [inline pkg/cluster/internal/create/kubeadm-init.go]: with respect to debugging e2e, would it make sense to add a `log.Info` message to indicate that we are using the LB port?

File: pkg/cluster/internal/create/kubeadm-init.go [hunk 3]
```diff
@@ -90,9 +91,7 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
 	// for the API server and exposed by the node
 
 	// retrives the random host where the API server is exposed
```
Comment (neolit123) [inline pkg/cluster/internal/create/kubeadm-init.go]: possibly remove this comment and add a comment above `getHostPort()`?

File: pkg/cluster/internal/create/kubeadm-init.go (hunk 2 above)
Comment (fabriziopandini) [reply pkg/cluster/internal/create/kubeadm-init.go]: done

File: pkg/cluster/internal/create/kubeadm-init.go [hunk 4]
```diff
@@ -155,6 +152,32 @@ func runKubeadmInit(ec *execContext, configNode *NodeReplica) error {
...
+		node = loadBalancerHandle
+		port = haproxy.ControlPlanePort
+	}
+
+	hostPort, err := node.Ports(port)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get Port from node: %s", node.String())
+	}
+
+	log.Infof("kubernetes cluster is accessible from the host using port %d exposed by node %s", hostPort, node.String())
```
Comment (BenTheElder) [inline pkg/cluster/internal/create/kubeadm-init.go]: tiny nit/suggestion: `s/is/should be/` -- we haven't actually verified that it's accessible, this kind of message can be frustrating when something isn't working 😅

**Comment by fabriziopandini** [comment]
/close 
Due to https://github.com/kubernetes-sigs/kind/pull/418

File: pkg/cluster/internal/create/actions/kubeadminit/init.go [hunk 5]
```diff
@@ -143,6 +141,33 @@ func (a *action) Execute(ctx *actions.ActionContext) error {
 	return nil
 }
 
+// getHostPort retrives the random host where the API server/the control plane is exposed
+func getHostPort(allNodes []nodes.Node) (int, error) {
```
Comment (BenTheElder) [inline pkg/cluster/internal/create/actions/kubeadminit/init.go]: :facepalm: I knew this problem was familiar in #418 ...

Thanks for this. Sorry it didn't get in sooner.

**Pull request closed without merging**